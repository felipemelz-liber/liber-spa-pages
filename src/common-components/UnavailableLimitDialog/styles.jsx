import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
