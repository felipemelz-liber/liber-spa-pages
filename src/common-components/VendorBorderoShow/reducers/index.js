import vendorBorderoShow from './VendorBorderoShowReducer';

export default {
  vendorBorderoShow,
};
