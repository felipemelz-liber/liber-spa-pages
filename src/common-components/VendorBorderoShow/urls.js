const domain = 'http://api.liber.local:3002'; // eslint-disable-line

const VENDORS = '/api/v1/admin/vendors';

export const VENDOR_BORDERO_SHOW_URL = `${domain}/api/v1/fornecedor/adiantamentos`;
export const VENDOR_BORDERO_INVOICES_URL = `${domain}/api/v1/fornecedor/adiantamentos/:id/titulos`;
export const CANCEL_PAGE_URL = `/fornecedor/antecipacoes/cancel`;
export const ADMIN_VENDOR_BORDERO_SHOW_URL = `${domain}${VENDORS}/:id/anticipations/:bordero_id`;
export const ADMIN_VENDOR_BORDERO_INVOICES_URL = `${domain}${VENDORS}/:id/anticipations/:bordero_id/invoices`;
