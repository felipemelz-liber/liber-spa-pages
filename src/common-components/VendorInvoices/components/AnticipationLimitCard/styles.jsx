import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
