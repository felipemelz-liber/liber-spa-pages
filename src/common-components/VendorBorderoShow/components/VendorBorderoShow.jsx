import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { CompleteTable } from 'liber-components';
import Icon from '@mdi/react';
import { mdiArrowLeft } from '@mdi/js';
import {
  Container,
  ScreenTitleContainer,
  ScreenTitle,
  LoadingContainer,
  Bunny,
  Button,
  CancelButton,
} from './VendorBorderoShow.styles';
import StatusLabel from './StatusLabel';
import ExpansibleCard from './ExpansibleCard/ExpansibleCard';
import CardContent from './CardContent/CardContent';
import {
  fetchSummary,
  fetchInvoices,
  previousPage,
  nextPage,
  setListing,
  setPage,
} from '../actions/VendorBorderoShowActions';
import { BORDERO_STATUS_MAP, getInvoicesColumns } from '../utils';
import { Row } from './CardContent/CardContent.styles';
import BorderoInvoicesFilters from './Filters/BorderoInvoicesFilters';
import InvoiceDialog from '../../../views/invoice-dialog/InvoiceDialog';
import CancellationDialog from '../../CancellationDialog/CancellationDialog';
import { CANCEL_PAGE_URL } from '../urls';
import usePageState from '../../../hooks/usePageState';
import HideFromAdmin from './HideFromAdmin/HideFromAdmin';
import { dispatchDataLayerEvent } from '../../../vendor/Utils';

export const VendorBorderoShow = ({
  status,
  onFetchSummary,
  summaryLoading,
  onFetchInvoices,
  invoices,
  pagination,
  filters,
  invoicesLoading,
  onNextPage,
  onPreviousPage,
  goToPage,
  onSetListing,
  userId,
  operators,
  borderoIds,
  hasDocument,
  token,
  isAdmin,
  vendorId,
}) => {
  const { current, pages: pageTotal, count: invoiceTotal, per } = pagination;

  const [invoiceShowOpen, setInvoiceShowOpen] = useState(false);
  const [invoiceId, setInvoiceId] = useState(null);
  const [pageState, setPageState, onKeyDownPage] = usePageState(current, pageTotal, goToPage);
  const [dialogOpen, setDialogOpen] = useState(false);
  const [borderoId] = borderoIds;

  React.useEffect(() => {
    onFetchSummary();
  }, []);

  React.useEffect(() => {
    onFetchInvoices();
  }, [current, per, filters]);

  const getInterval = () => {
    const end = current * per;
    const start = (current - 1) * per + 1;
    return { start: String(start), end: String(end > invoiceTotal ? invoiceTotal : end) };
  };

  const handleBack = () => window.history.back();

  const needUserConfirmation = () => {
    let need = false;
    operators.forEach(({ id, userHasSigned }) => {
      if (userId === id && !userHasSigned && status !== 'canceled') {
        need = true;
      }
    });
    return need;
  };

  const handleConfirmation = () => {
    dispatchDataLayerEvent('showBorderoConfirmarAntecipacao', isAdmin);

    const ids = borderoIds.map(id => `ids[]=${id}`).join('&');

    window.location.href = `/fornecedor/antecipacoes/sign?${ids}${
      hasDocument ? '&generated=true' : ''
    }`;
  };

  const handleCancellationConfirm = () => {
    dispatchDataLayerEvent('showBorderoCancelarAntecipacao', isAdmin);

    window.location.href = `${CANCEL_PAGE_URL}?id=${borderoIds[0]}`;
  };

  return (
    <>
      <Container>
        <ScreenTitleContainer>
          <HideFromAdmin>
            <Icon onClick={handleBack} path={mdiArrowLeft} />
          </HideFromAdmin>
          <ScreenTitle>Antecipação</ScreenTitle>
          <Row justify="space-between">
            {status && BORDERO_STATUS_MAP[status] && (
              <StatusLabel status={BORDERO_STATUS_MAP[status].type}>
                {BORDERO_STATUS_MAP[status].label}
              </StatusLabel>
            )}
            {operators && needUserConfirmation() && (
              <HideFromAdmin>
                <div>
                  <CancelButton onClick={() => setDialogOpen(true)}>CANCELAR</CancelButton>
                  <Button onClick={handleConfirmation}>Confirmar</Button>
                </div>
              </HideFromAdmin>
            )}
          </Row>
        </ScreenTitleContainer>
        {!summaryLoading && (
          <ExpansibleCard
            buttonText={{
              closed: 'Mostrar Confirmação e Pagamento',
              open: 'Ocultar Confirmação e Pagamento',
            }}
            collapsedHeight={264}
            showButton={status !== 'created'}
            isAdmin={isAdmin}
          >
            <CardContent />
          </ExpansibleCard>
        )}
        <Row height={40} />
        <CompleteTable
          items={invoicesLoading ? [] : invoices}
          columns={getInvoicesColumns({
            borderoStatus: status,
            setInvoiceId,
            setInvoiceShowOpen,
            isAdmin,
          })}
          filters={<BorderoInvoicesFilters isAdmin={isAdmin} />}
          emptyState={
            invoicesLoading ? (
              <LoadingContainer>
                <Bunny />
              </LoadingContainer>
            ) : undefined
          }
          paginationProps={{
            currentPage: String(pageState),
            pageTotal: String(pageTotal),
            onClickNext: onNextPage,
            onClickPrev: onPreviousPage,
            onChangePage: ({ target: { value } }) => {
              setPageState(value);
            },
            onKeyDownPage,
            hasListing: true,
            listingProps: {
              listing: String(per),
              total: String(invoiceTotal),
              showInterval: getInterval(pagination),
              onSelectListing: onSetListing,
            },
          }}
        />
        <InvoiceDialog
          token={token}
          open={invoiceShowOpen}
          invoiceId={invoiceId}
          borderoId={borderoId}
          vendorId={vendorId}
          isAdmin={isAdmin}
          onClose={() => setInvoiceShowOpen(false)}
        />
      </Container>
      <CancellationDialog
        open={dialogOpen}
        onClose={() => {
          setDialogOpen(false);
        }}
        onConfirm={handleCancellationConfirm}
      />
    </>
  );
};

VendorBorderoShow.propTypes = {
  status: PropTypes.string,
  onFetchSummary: PropTypes.func,
  summaryLoading: PropTypes.bool,
  onFetchInvoices: PropTypes.func,
  invoices: PropTypes.arrayOf(PropTypes.shape({})),
  pagination: PropTypes.shape({
    current: PropTypes.number,
    previous: PropTypes.number,
    next: PropTypes.number,
    per: PropTypes.number,
    pages: PropTypes.number,
    count: PropTypes.number,
  }),
  filters: PropTypes.shape({}),
  invoicesLoading: PropTypes.bool,
  onNextPage: PropTypes.func,
  onPreviousPage: PropTypes.func,
  goToPage: PropTypes.func,
  onSetListing: PropTypes.func,
  userId: PropTypes.number,
  operators: PropTypes.arrayOf(PropTypes.shape({})),
  borderoIds: PropTypes.arrayOf(PropTypes.number),
  hasDocument: PropTypes.bool,
  token: PropTypes.string,
  isAdmin: PropTypes.bool,
  vendorId: PropTypes.number,
};

VendorBorderoShow.defaultProps = {
  status: null,
  onFetchSummary: () => {},
  summaryLoading: false,
  onFetchInvoices: () => [],
  invoices: [],
  pagination: {
    current: 1,
    previous: null,
    next: null,
    per: 10,
    pages: 1,
    count: 0,
  },
  filters: {},
  invoicesLoading: false,
  onNextPage: () => null,
  onPreviousPage: () => null,
  goToPage: () => null,
  onSetListing: () => null,
  userId: null,
  operators: [],
  borderoIds: [],
  hasDocument: false,
  token: '',
  isAdmin: false,
  vendorId: null,
};

const mapStateToProps = ({
  vendorBorderoShow: {
    summary: { status },
    summaryLoading,
    invoices,
    pagination,
    filters,
    invoicesLoading,
    operator: { id: userId },
    summary: { operators, invoiceTransferAgreementUrl: hasDocument },
    borderoIds,
    token,
    isAdmin,
    fornecedorId: vendorId,
  },
}) => ({
  status,
  summaryLoading,
  invoices,
  pagination,
  filters,
  invoicesLoading,
  userId,
  operators,
  borderoIds,
  hasDocument: Boolean(hasDocument),
  token,
  isAdmin,
  vendorId,
});

const mapDispatchToProps = {
  onFetchSummary: fetchSummary,
  onFetchInvoices: fetchInvoices,
  onNextPage: nextPage,
  onPreviousPage: previousPage,
  onSetListing: setListing,
  goToPage: setPage,
};

export default connect(mapStateToProps, mapDispatchToProps)(VendorBorderoShow);
