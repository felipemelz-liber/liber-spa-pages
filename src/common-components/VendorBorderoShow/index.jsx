import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import hydrateStore from '../../views/store';
import VendorBorderoShow from './components/VendorBorderoShow';

const VendorBorderoShowComponent = ({ token, fornecedorId, operator, borderoIds, isAdmin }) => (
  <>
    <Provider
      store={hydrateStore({
        vendorBorderoShow: { token, fornecedorId, operator, borderoIds, isAdmin },
        invoiceDialog: { isAdmin },
      })}
    >
      <VendorBorderoShow />
    </Provider>
  </>
);

VendorBorderoShowComponent.propTypes = {
  token: PropTypes.string.isRequired,
  fornecedorId: PropTypes.number.isRequired,
  operator: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    cpf: PropTypes.string,
  }).isRequired,
  borderoIds: PropTypes.arrayOf(PropTypes.number).isRequired,
  isAdmin: PropTypes.bool,
};

VendorBorderoShowComponent.defaultProps = {
  isAdmin: false,
};

export default VendorBorderoShowComponent;
