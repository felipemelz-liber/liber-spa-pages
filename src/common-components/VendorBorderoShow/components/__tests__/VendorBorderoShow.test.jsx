import React from 'react';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Icon from '@mdi/react';
import { VendorBorderoShow } from '../VendorBorderoShow';
import { Button } from '../VendorBorderoShow.styles';

describe('VendorBorderoShow component tests', () => {
  configure({ adapter: new Adapter() });

  const testCases = [
    {
      label: 'ConcluÃ­da',
      status: 'finished',
    },
    {
      label: 'Realizando Pagamento',
      status: 'signed',
    },
  ];

  it('should render correctly without props', () => {
    const wrapper = shallow(<VendorBorderoShow />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with summary loading', () => {
    const wrapper = shallow(<VendorBorderoShow summaryLoading />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  testCases.forEach(({ label, status }) => {
    it(`should render correctly with ${label} status`, () => {
      const wrapper = shallow(<VendorBorderoShow status={status} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  it('should render correctly with summary loading', () => {
    const onFetchSummary = sinon.spy();
    const useEffectStub = sinon.stub(React, 'useEffect');
    useEffectStub.callsArg(0);
    shallow(<VendorBorderoShow onFetchSummary={onFetchSummary} />);
    expect(onFetchSummary.called).toBeTruthy();
  });

  it('should call window.history.back() when clicking on back icon button', () => {
    const backStub = sinon.stub(window.history, 'back');
    const wrapper = shallow(<VendorBorderoShow />);
    const backButton = wrapper.find(Icon);
    backButton.simulate('click');
    expect(backStub.called).toBeTruthy();
  });

  it('should have confirmation button only if user hasnt signed yet', () => {
    const operatorsSigned = [
      {
        id: 77,
        name: 'Operator',
        userHasSigned: true,
      },
    ];

    const operatorsNotSigned = [
      {
        id: 77,
        name: 'Operator',
        userHasSigned: false,
      },
    ];

    const signedWrapper = shallow(<VendorBorderoShow userId={77} operators={operatorsSigned} />);
    let confirmationButton = signedWrapper.find(Button);
    expect(confirmationButton.length).toBe(0);

    confirmationButton = null;

    const notSignedWrapper = shallow(
      <VendorBorderoShow userId={77} operators={operatorsNotSigned} />,
    );
    confirmationButton = notSignedWrapper.find(Button);
    expect(confirmationButton.length).toBe(1);
  });
});
