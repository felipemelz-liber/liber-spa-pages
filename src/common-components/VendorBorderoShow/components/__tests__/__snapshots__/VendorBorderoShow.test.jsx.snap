// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorBorderoShow component tests should render correctly with Concluída status 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Connect(HideFromAdmin)>
        <Icon
          color="currentColor"
          horizontal={false}
          onClick={[Function]}
          path="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
          rotate={0}
          size={null}
          spin={false}
          vertical={false}
        />
      </Connect(HideFromAdmin)>
      <styled.span>
        Antecipação
      </styled.span>
      <styled.div
        justify="space-between"
      >
        <StatusLabel
          color=""
          status="positive"
        >
          Concluída
        </StatusLabel>
      </styled.div>
    </styled.div>
    <ExpansibleCard
      buttonText={
        Object {
          "closed": "Mostrar Confirmação e Pagamento",
          "open": "Ocultar Confirmação e Pagamento",
        }
      }
      collapsedHeight={264}
      isAdmin={false}
      showButton={true}
    >
      <Connect(CardContent) />
    </ExpansibleCard>
    <styled.div
      height={40}
    />
    <CompleteTable
      actions={null}
      allChecked={false}
      borderBottom={false}
      columns={
        Array [
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": <React.Fragment>
                 Status
            </React.Fragment>,
            "name": "status",
            "width": 17,
          },
          Object {
            "align": "right",
            "label": "Número",
            "name": "number",
            "width": 15,
          },
          Object {
            "align": "right",
            "getCellValue": [Function],
            "label": "Valor Antecipado (R$)",
            "name": "netValue",
            "width": 16,
          },
          Object {
            "align": "right",
            "label": "CNPJ",
            "name": "vendorCnpj",
            "width": 15,
          },
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": "Investidor",
            "name": "funder",
            "width": 14,
          },
          Object {
            "align": "center",
            "getCellValue": [Function],
            "label": "Ações",
            "name": "detailsPath",
            "width": 7,
          },
        ]
      }
      emptyState={<Empty />}
      filterBorderTop={true}
      filters={
        <Memo(Connect(BorderoInvoicesFilters))
          isAdmin={false}
        />
      }
      headerBorderTop={true}
      hiddenColumns={Array []}
      isItemSelectable={[Function]}
      isItemSelected={[Function]}
      items={Array []}
      onSelect={[Function]}
      onSelectAll={null}
      paginationProps={
        Object {
          "currentPage": "1",
          "hasListing": true,
          "listingProps": Object {
            "listing": "10",
            "onSelectListing": [Function],
            "showInterval": Object {
              "end": "0",
              "start": "1",
            },
            "total": "0",
          },
          "onChangePage": [Function],
          "onClickNext": [Function],
          "onClickPrev": [Function],
          "onKeyDownPage": [Function],
          "pageTotal": "1",
        }
      }
      selectionControl={null}
      sortingProps={
        Object {
          "onSort": [Function],
          "sortedColumns": false,
          "sorting": Array [],
        }
      }
      tooltipContent={[Function]}
      weightedColumns={false}
    />
    <Connect(InvoiceDialog)
      invoiceId={null}
      isAdmin={false}
      onClose={[Function]}
      open={false}
      token=""
      vendorId={null}
    />
  </styled.div>
  <CancellationDialog
    onClose={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`VendorBorderoShow component tests should render correctly with Realizando Pagamento status 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Connect(HideFromAdmin)>
        <Icon
          color="currentColor"
          horizontal={false}
          onClick={[Function]}
          path="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
          rotate={0}
          size={null}
          spin={false}
          vertical={false}
        />
      </Connect(HideFromAdmin)>
      <styled.span>
        Antecipação
      </styled.span>
      <styled.div
        justify="space-between"
      >
        <StatusLabel
          color=""
          status="neutral"
        >
          Realizando Pagamento
        </StatusLabel>
      </styled.div>
    </styled.div>
    <ExpansibleCard
      buttonText={
        Object {
          "closed": "Mostrar Confirmação e Pagamento",
          "open": "Ocultar Confirmação e Pagamento",
        }
      }
      collapsedHeight={264}
      isAdmin={false}
      showButton={true}
    >
      <Connect(CardContent) />
    </ExpansibleCard>
    <styled.div
      height={40}
    />
    <CompleteTable
      actions={null}
      allChecked={false}
      borderBottom={false}
      columns={
        Array [
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": <React.Fragment>
                 Status
            </React.Fragment>,
            "name": "status",
            "width": 17,
          },
          Object {
            "align": "right",
            "label": "Número",
            "name": "number",
            "width": 15,
          },
          Object {
            "align": "right",
            "getCellValue": [Function],
            "label": "Valor Solicitado (R$)",
            "name": "netValue",
            "width": 16,
          },
          Object {
            "align": "right",
            "label": "CNPJ",
            "name": "vendorCnpj",
            "width": 15,
          },
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": "Investidor",
            "name": "funder",
            "width": 14,
          },
          Object {
            "align": "center",
            "getCellValue": [Function],
            "label": "Ações",
            "name": "detailsPath",
            "width": 7,
          },
        ]
      }
      emptyState={<Empty />}
      filterBorderTop={true}
      filters={
        <Memo(Connect(BorderoInvoicesFilters))
          isAdmin={false}
        />
      }
      headerBorderTop={true}
      hiddenColumns={Array []}
      isItemSelectable={[Function]}
      isItemSelected={[Function]}
      items={Array []}
      onSelect={[Function]}
      onSelectAll={null}
      paginationProps={
        Object {
          "currentPage": "1",
          "hasListing": true,
          "listingProps": Object {
            "listing": "10",
            "onSelectListing": [Function],
            "showInterval": Object {
              "end": "0",
              "start": "1",
            },
            "total": "0",
          },
          "onChangePage": [Function],
          "onClickNext": [Function],
          "onClickPrev": [Function],
          "onKeyDownPage": [Function],
          "pageTotal": "1",
        }
      }
      selectionControl={null}
      sortingProps={
        Object {
          "onSort": [Function],
          "sortedColumns": false,
          "sorting": Array [],
        }
      }
      tooltipContent={[Function]}
      weightedColumns={false}
    />
    <Connect(InvoiceDialog)
      invoiceId={null}
      isAdmin={false}
      onClose={[Function]}
      open={false}
      token=""
      vendorId={null}
    />
  </styled.div>
  <CancellationDialog
    onClose={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`VendorBorderoShow component tests should render correctly with summary loading 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Connect(HideFromAdmin)>
        <Icon
          color="currentColor"
          horizontal={false}
          onClick={[Function]}
          path="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
          rotate={0}
          size={null}
          spin={false}
          vertical={false}
        />
      </Connect(HideFromAdmin)>
      <styled.span>
        Antecipação
      </styled.span>
      <styled.div
        justify="space-between"
      />
    </styled.div>
    <styled.div
      height={40}
    />
    <CompleteTable
      actions={null}
      allChecked={false}
      borderBottom={false}
      columns={
        Array [
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": <React.Fragment>
                 Status
            </React.Fragment>,
            "name": "status",
            "width": 17,
          },
          Object {
            "align": "right",
            "label": "Número",
            "name": "number",
            "width": 15,
          },
          Object {
            "align": "right",
            "getCellValue": [Function],
            "label": "Valor Solicitado (R$)",
            "name": "netValue",
            "width": 16,
          },
          Object {
            "align": "right",
            "label": "CNPJ",
            "name": "vendorCnpj",
            "width": 15,
          },
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": "Investidor",
            "name": "funder",
            "width": 14,
          },
          Object {
            "align": "center",
            "getCellValue": [Function],
            "label": "Ações",
            "name": "detailsPath",
            "width": 7,
          },
        ]
      }
      emptyState={<Empty />}
      filterBorderTop={true}
      filters={
        <Memo(Connect(BorderoInvoicesFilters))
          isAdmin={false}
        />
      }
      headerBorderTop={true}
      hiddenColumns={Array []}
      isItemSelectable={[Function]}
      isItemSelected={[Function]}
      items={Array []}
      onSelect={[Function]}
      onSelectAll={null}
      paginationProps={
        Object {
          "currentPage": "1",
          "hasListing": true,
          "listingProps": Object {
            "listing": "10",
            "onSelectListing": [Function],
            "showInterval": Object {
              "end": "0",
              "start": "1",
            },
            "total": "0",
          },
          "onChangePage": [Function],
          "onClickNext": [Function],
          "onClickPrev": [Function],
          "onKeyDownPage": [Function],
          "pageTotal": "1",
        }
      }
      selectionControl={null}
      sortingProps={
        Object {
          "onSort": [Function],
          "sortedColumns": false,
          "sorting": Array [],
        }
      }
      tooltipContent={[Function]}
      weightedColumns={false}
    />
    <Connect(InvoiceDialog)
      invoiceId={null}
      isAdmin={false}
      onClose={[Function]}
      open={false}
      token=""
      vendorId={null}
    />
  </styled.div>
  <CancellationDialog
    onClose={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`VendorBorderoShow component tests should render correctly without props 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <Connect(HideFromAdmin)>
        <Icon
          color="currentColor"
          horizontal={false}
          onClick={[Function]}
          path="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
          rotate={0}
          size={null}
          spin={false}
          vertical={false}
        />
      </Connect(HideFromAdmin)>
      <styled.span>
        Antecipação
      </styled.span>
      <styled.div
        justify="space-between"
      />
    </styled.div>
    <ExpansibleCard
      buttonText={
        Object {
          "closed": "Mostrar Confirmação e Pagamento",
          "open": "Ocultar Confirmação e Pagamento",
        }
      }
      collapsedHeight={264}
      isAdmin={false}
      showButton={true}
    >
      <Connect(CardContent) />
    </ExpansibleCard>
    <styled.div
      height={40}
    />
    <CompleteTable
      actions={null}
      allChecked={false}
      borderBottom={false}
      columns={
        Array [
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": <React.Fragment>
                 Status
            </React.Fragment>,
            "name": "status",
            "width": 17,
          },
          Object {
            "align": "right",
            "label": "Número",
            "name": "number",
            "width": 15,
          },
          Object {
            "align": "right",
            "getCellValue": [Function],
            "label": "Valor Solicitado (R$)",
            "name": "netValue",
            "width": 16,
          },
          Object {
            "align": "right",
            "label": "CNPJ",
            "name": "vendorCnpj",
            "width": 15,
          },
          Object {
            "align": "left",
            "getCellValue": [Function],
            "label": "Investidor",
            "name": "funder",
            "width": 14,
          },
          Object {
            "align": "center",
            "getCellValue": [Function],
            "label": "Ações",
            "name": "detailsPath",
            "width": 7,
          },
        ]
      }
      emptyState={<Empty />}
      filterBorderTop={true}
      filters={
        <Memo(Connect(BorderoInvoicesFilters))
          isAdmin={false}
        />
      }
      headerBorderTop={true}
      hiddenColumns={Array []}
      isItemSelectable={[Function]}
      isItemSelected={[Function]}
      items={Array []}
      onSelect={[Function]}
      onSelectAll={null}
      paginationProps={
        Object {
          "currentPage": "1",
          "hasListing": true,
          "listingProps": Object {
            "listing": "10",
            "onSelectListing": [Function],
            "showInterval": Object {
              "end": "0",
              "start": "1",
            },
            "total": "0",
          },
          "onChangePage": [Function],
          "onClickNext": [Function],
          "onClickPrev": [Function],
          "onKeyDownPage": [Function],
          "pageTotal": "1",
        }
      }
      selectionControl={null}
      sortingProps={
        Object {
          "onSort": [Function],
          "sortedColumns": false,
          "sorting": Array [],
        }
      }
      tooltipContent={[Function]}
      weightedColumns={false}
    />
    <Connect(InvoiceDialog)
      invoiceId={null}
      isAdmin={false}
      onClose={[Function]}
      open={false}
      token=""
      vendorId={null}
    />
  </styled.div>
  <CancellationDialog
    onClose={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;
